ddtrace/internal/writer.py

```diff
@@ -164,22 +167,25 @@ def put(self, item):
 
     def _update_stats(self, item):
         # self.mutex needs to be locked to make sure we don't lose data when resetting
-        self.enqueued += 1
+        self.accepted += 1
         if hasattr(item, '__len__'):
             item_length = len(item)
         else:
             item_length = 1
-        self.enqueued_lengths += item_length
+        self.accepted_lengths += item_length
+        self.accepted_size += sizeof.sizeof(item)
 
     def reset_stats(self):
         """Reset the stats to 0.
 
-        :return: The current value of dropped, enqueued and enqueued_lengths.
+        :return: The current value of dropped, accepted and accepted_lengths.
         """
         with self.mutex:
-            dropped, enqueued, enqueued_lengths = self.dropped, self.enqueued, self.enqueued_lengths
-            self.dropped, self.enqueued, self.enqueued_lengths = 0, 0, 0
-        return dropped, enqueued, enqueued_lengths
+            dropped, accepted, accepted_lengths, accepted_size = (
```

Comment:
would a namedtuple help here?

Reply:
It's a good idea. I'll keep it that way for now but if it grows, I'll use one. :)